packages/graphback-core/src/relationships/RelationshipMetadataBuilder.ts

```diff
@@ -139,109 +149,113 @@ export class RelationshipMetadataBuilder {
-    this.validateOneToManyRelationship(ownerType.name, field);
+    return field;
+  }
 
-    const metadata = parseRelationshipAnnotation(field.description);
+  private addOneToMany(ownerType: GraphQLObjectType, field: GraphQLField<any, any>, oneToManyAnnotation: RelationshipAnnotation) {
+    this.validateOneToManyRelationship(ownerType.name, field, oneToManyAnnotation);
 
     // skip - relationship with key annotation will be generated at later stage
-    if (!metadata.key) {
+    if (!oneToManyAnnotation.key) {
       return;
     }
 
     const relationType = getNamedType(field.type) as GraphQLObjectType;
-    const foreignKeyField = metadata.key || transformForeignKeyName(metadata.field);
 
     const oneToMany: FieldRelationshipMetadata = {
       kind: 'oneToMany',
       owner: ownerType,
       ownerField: field,
       relationType,
-      relationFieldName: metadata.field,
-      relationForeignKey: foreignKeyField
+      relationFieldName: oneToManyAnnotation.field,
+      relationForeignKey: oneToManyAnnotation.key
     };
 
     this.relationships.push(oneToMany);
   }
 
-  private addManyToOne(ownerType: GraphQLObjectType, field: GraphQLField<any, any>) {
-    this.validateManyToOneField(ownerType.name, field);
-
-    const metadata = parseRelationshipAnnotation(field.description);
+  private addManyToOne(ownerType: GraphQLObjectType, field: GraphQLField<any, any>, manyToOneAnnotation: RelationshipAnnotation) {
+    this.validateManyToOneField(ownerType.name, field, manyToOneAnnotation);
 
     // skip - relationship with key annotation will be generated at later stage
-    if (!metadata.key) {
+    if (!manyToOneAnnotation.key) {
       return;
     }
 
     const relationType = getNamedType(field.type) as GraphQLObjectType;
-    const columnField = metadata.key || transformForeignKeyName(field.name);
 
     const manyToOne: FieldRelationshipMetadata = {
       kind: 'manyToOne',
       owner: ownerType,
       ownerField: field,
       relationType,
-      relationFieldName: metadata.field,
-      relationForeignKey: columnField
+      relationFieldName: manyToOneAnnotation.field,
+      relationForeignKey: manyToOneAnnotation.key
     }
 
     this.relationships.push(manyToOne);
   }
 
-  private addOneToOne(ownerType: GraphQLObjectType, field: GraphQLField<any, any>) {
-    this.validateOneToOneRelationship(ownerType.name, field);
-
-    const metadata = parseRelationshipAnnotation(field.description);
+  private addOneToOne(ownerType: GraphQLObjectType, field: GraphQLField<any, any>, oneToOneAnnotation: RelationshipAnnotation) {
+    this.validateOneToOneRelationship(ownerType.name, field, oneToOneAnnotation);
 
     // skip - relationship with key mapping will be generated at a later stage
-    if (!metadata.key) {
+    if (!oneToOneAnnotation.key) {
       return;
     }
 
     const relationType = getNamedType(field.type) as GraphQLObjectType;
 
-    const columnField = metadata.key || transformForeignKeyName(field.name);
```

Comment:
This enables shorter annotations by assuming a default foreign key column name.

Reply:
Yes, except that because of the following conditions: 
 - https://github.com/aerogear/graphback/blob/6f80afa191444414d8d89d14a8198f2c3ae4d12a/packages/graphback-core/src/relationships/RelationshipMetadataBuilder.ts#L246
- https://github.com/aerogear/graphback/blob/6f80afa191444414d8d89d14a8198f2c3ae4d12a/packages/graphback-core/src/relationships/RelationshipMetadataBuilder.ts#L224
- https://github.com/aerogear/graphback/blob/6f80afa191444414d8d89d14a8198f2c3ae4d12a/packages/graphback-core/src/relationships/RelationshipMetadataBuilder.ts#L202

The second expression will never be evaluated.